pkg/nmi/server/server.go

```diff
@@ -189,27 +189,34 @@ func (fn appHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}()
 	rw := newResponseWriter(w)
-	fn(logger, rw, r)
+	ns := fn(logger, rw, r)
 	latency := time.Since(start)
 	logger.Infof("Status (%d) took %d ns", rw.statusCode, latency.Nanoseconds())
 
-	namespace, _ := parseRequestHeader(r)
 	_, resource := parseRequestClientIDAndResource(r)
 
 	if appHandlerReporter != nil {
 		appHandlerReporter.ReportOperationAndStatus(
 			r.URL.Path,
 			strconv.Itoa(rw.statusCode),
-			namespace,
+			ns,
 			resource,
-			metrics.NodeManagedIdentityOperationsDurationM.M(metrics.SinceInSeconds(start)))
+			metrics.NMIOperationsDurationM.M(metrics.SinceInSeconds(start)))
 	}
 }
 
-func (s *Server) hostHandler(logger *log.Entry, w http.ResponseWriter, r *http.Request) {
+func (s *Server) hostHandler(logger *log.Entry, w http.ResponseWriter, r *http.Request) (ns string) {
```

Comment:
Are we returning namespaces from this ? Also, does not look like all return paths return the namespace ?